encoding/codecv2/codecv2.go

```diff
@@ -183,6 +184,15 @@ func ConstructBlobPayload(chunks []*encoding.Chunk, useMockTxData bool) (*kzg484
 		return nil, common.Hash{}, nil, err
 	}
 
+	// Only apply this check when the uncompressed batch data has exceeded 128 KiB.
+	if len(blobBytes) > 131072 {
+		// Check compressed data compatibility.
+		if err = encoding.CheckCompressedDataCompatibility(blobBytes); err != nil {
+			log.Error("ConstructBlobPayload: compressed data compatibility check failed", "err", err, "uncompressedBlobBytes", hex.EncodeToString(blobBytes), "compressedBlobBytes", hex.EncodeToString(compressedBlobBytes))
+			return nil, common.Hash{}, nil, &encoding.CompressedDataCompatibilityError{Err: err}
+		}
+	}
```

Comment:
@colinlyguo when does this scenario occur? `blobBytes < 4096 * 31` will always hold, right? So isn't this check redundant?

We should instead have this check for `batchBytes`, which is already done [here](https://github.com/scroll-tech/da-codec/pull/17/files#diff-626a61d73695a9154e13d35f43a9633078eacc8ce3e1cb8a290e8c0723a0a0dcR319-R325)

Reply:
`blobBytes` here is uncompressed (as opposed to `compressedBlobBytes` a few lines above).